tests/e2e/mapper/bluetooth/bluetooth_test.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bluetooth
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/kubeedge/kubeedge/cloud/pkg/devicecontroller/apis/devices/v1alpha1"
+	"github.com/kubeedge/kubeedge/device/bluetooth_mapper/scheduler"
+	"github.com/kubeedge/kubeedge/device/bluetooth_mapper/watcher"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"reflect"
+	"time"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubeedge/kubeedge/edge/test/integration/utils/common"
+	"github.com/kubeedge/kubeedge/edge/test/integration/utils/helpers"
+	"github.com/kubeedge/kubeedge/tests/e2e/utils"
+)
+
+var TokenClient Token
+var ClientOpts *MQTT.ClientOptions
+var Client MQTT.Client
+var timesSpecified int
+var timesExecuted int
+
+type ScheduleResult struct {
+	EventName   string
+	TimeStamp   int64
+	EventResult string
+}
+
+var scheduleResult ScheduleResult
+var dataConverted bool
+var readWrittenData bool
+
+// DataConversion checks whether data is properly as expected by the data converter.
+func DataConversion(client MQTT.Client, message MQTT.Message) {
+	topic := "$ke/device/bluetooth-mapper/mock-temp-sensor-instance/scheduler/result"
+	expectedTemp := "32.375000"
+	if message.Topic() == topic {
+		dataConverted = false
```

Comment:
it is better to move dataConverted = false at the beginning of the function ?
If this function is called twice, and the if condition above is not satisfied, the value will still be true

Reply:
Updtaed